packages/compass-e2e-tests/helpers/commands/connect-with-connection-form.ts

```diff
@@ -56,150 +40,33 @@ export async function connectWithConnectionForm(
-      ? 'X509'
-      : username || password
-      ? 'MONGODB'
-      : 'NONE';
-
-  const authStrategyInputComponent = await browser.$(
-    Selectors.ConnectionFormInputAuthStrategy
-  );
-  await authStrategyInputComponent.selectByAttribute('value', authStrategy);
-
-  if (typeof username !== 'undefined') {
-    const kerberosPrincipalInputElement = await browser.$(
-      Selectors.ConnectionFormInputKerberosPrincipal
-    );
-    const ldapUsernameInputElement = await browser.$(
-      Selectors.ConnectionFormInputLDAPUsername
-    );
-    // TODO: No point in having different `name`s in UI, they are not used for
-    // anything and all those map to `username` in driver options anyway
-    if (await kerberosPrincipalInputElement.isDisplayed()) {
-      const element = await browser.$(
-        Selectors.ConnectionFormInputKerberosPrincipal
-      );
-      await element.setValue(username);
-    } else if (await ldapUsernameInputElement.isDisplayed()) {
-      const element = await browser.$(
-        Selectors.ConnectionFormInputLDAPUsername
-      );
-      await element.setValue(username);
-    } else {
-      const element = await browser.$(Selectors.ConnectionFormInputUsername);
-      await element.setValue(username);
-    }
-  }
-
-  if (typeof password !== 'undefined') {
-    const ldapPasswordInputElement = await browser.$(
-      Selectors.ConnectionFormInputLDAPPassword
-    );
-    if (await ldapPasswordInputElement.isDisplayed()) {
-      const element = await browser.$(
-        Selectors.ConnectionFormInputLDAPPassword
-      );
-      await element.setValue(password);
-    } else {
-      const element = await browser.$(Selectors.ConnectionFormInputPassword);
-      await element.setValue(password);
-    }
-  }
-
-  if (typeof gssapiServiceName !== 'undefined') {
-    const element = await browser.$('[name="kerberos-service-name"]');
-    await element.setValue(gssapiServiceName);
-  }
-
-  await browser.clickVisible('#More_Options');
-
-  if (typeof replicaSet !== 'undefined') {
-    const element = await browser.$(Selectors.ConnectionFormInputReplicaSet);
-    await element.setValue(replicaSet);
-  }
-
-  const sslMethod =
-    tlsAllowInvalidHostnames === true || sslValidate === false
-      ? 'UNVALIDATED'
-      : typeof tlsCAFile !== 'undefined' &&
-        typeof tlsCertificateKeyFile !== 'undefined'
-      ? 'ALL'
-      : typeof tlsCAFile !== 'undefined'
-      ? 'SERVER'
-      : /mongodb.net$/.test(host)
-      ? 'SYSTEMCA'
-      : 'NONE';
-
-  const sslMethodInputComponent = await browser.$(
-    Selectors.ConnectionFormInputSSLMethod
-  );
-  await sslMethodInputComponent.selectByAttribute('value', sslMethod);
-
-  if (['ALL', 'SERVER'].includes(sslMethod)) {
-    // TODO: Can be implemented after https://github.com/mongodb-js/compass/pull/2380
-    throw new Error("Can't test connections that use SSL");
+    await browser.click(Selectors.ConnectionFormInputSrvRecord);
```

Comment:
What's wrong with browser.clickVisible?

Reply:
I'm surprised it would allow you to click on an invisible thing, though. You really should be clicking on a visible thing the user could be interacting with ideally in the same way the user would.

Reply:
that's a good point, would that be something like `label = findTheLabelOfThatButton; await browser.clickVisible(label)`?

Reply:
clickVisible only takes a selector, but something like that. If you have to find an element with some alternative selector strategy, then just do label.click().

But generally please try and use clickVisible as that's where we put all the workarounds like waiting for things to be visible, waiting for animations to stop and now (shortly) maybe scrolling the thing into view.

Reply:
Done, worked great